lumigator/backend/backend/tracking/mlflow.py

```diff
@@ -420,17 +319,280 @@ def list_jobs(self, workflow_id: str):
+        for job_id in job_ids:
+            run = self._client.get_run(job_id)
+            for metric in run.data.metrics:
+                assert metric not in metrics
+                metrics[metric] = run.data.metrics[metric]
+
+        return metrics
+
+    def _compile_parameters(self, job_ids: list) -> dict:
+        """Take the individual parameters from each run and compile them into a single dict
+        for now, assert that each run has no overlapping parameters
+        """
+        parameters = {}
+        for job_id in job_ids:
+            run = self._client.get_run(job_id)
+            for parameter in run.data.params:
+                # if the parameter shows up in multiple runs, prepend the run_name to the key
+                # TODO: this is a hacky way to handle this,
+                #  we should come up with a better solution but at least it keeps the info
+                if parameter in parameters:
+                    parameters[f"{run.data.tags['mlflow.runName']}_{parameter}"] = run.data.params[parameter]
+                parameters[parameter] = run.data.params[parameter]
+        return parameters
+
+    def _find_workflows(self, experiment_id: str) -> list:
+        """Find all the workflows associated with an experiment."""
+        all_runs = self._client.search_runs(experiment_ids=[experiment_id])
+        # now organize the runs into workflows so that a nested run goes under the parent run
+        workflow_ids = []
+        for run in all_runs:
+            parent_id = run.data.tags.get(MLFLOW_PARENT_RUN_ID)
+            # if it has a parent id then it's a run, if not then it's a workflow
+            if parent_id is None:
+                workflow_ids.append(run.info.run_id)
+        return workflow_ids
+
+    def _get_ray_job_logs(self, ray_job_id: str):
+        """Get the logs for a Ray job."""
+        resp = requests.get(urljoin(settings.RAY_JOBS_URL, f"{ray_job_id}/logs"), timeout=5)  # 5 sec
+
+        if resp.status_code == HTTPStatus.NOT_FOUND:
+            loguru.logger.error(f"Upstream job logs not found: {resp.status_code}, error: {resp.text or ''}")
+            raise JobNotFoundError(UUID(ray_job_id), "Ray job logs not found") from None
+        elif resp.status_code != HTTPStatus.OK:
+            loguru.logger.error(
+                f"Unexpected status code getting job logs: {resp.status_code}, \
+                    error: {resp.text or ''}"
+            )
+            raise JobUpstreamError(ray_job_id, "Non OK status code retrieving Ray job information") from None
+
+        try:
+            metadata = json.loads(resp.text)
+            return JobLogsResponse(**metadata)
+        except json.JSONDecodeError as e:
+            loguru.logger.error(f"JSON decode error: {e}")
+            loguru.logger.error(f"Response text: {resp.text}")
+            raise JobUpstreamError(ray_job_id, "JSON decode error in Ray response") from e
+
+    def _generate_presigned_url(self, workflow_id: str) -> str:
```

Comment:
I think this should be the responsibility of the backend, and not the tracking client. Again, if it's a carryover please ignore it.

Reply:
It was part of the old code: 
https://github.com/mozilla-ai/lumigator/pull/1215/files#diff-f93ebfc8356d0b50af30b5f79b5060cfb6c1120edbe593810af7fd9a77e670d5L294-L304